.github/workflows/wheels.yml

```diff
@@ -5,9 +5,70 @@ on:
     tags:
       - '*'
 jobs:
+  build_wheels:
+    name: Build wheels
+    runs-on: ${{ matrix.os }}
+    environment: release
+    permissions:
+      id-token: write
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Since we're looking at this now, do you think there'd be a use in pinning the images to exact versions?  Certainly this change changes the macOS image from an 11.x image to a 12.x image - I don't know if that'll have much effect.

Reply:
Oh, that's a fair point. I don't think there is an issue with using 12.x here, but it's probably best to be conservative on the versions for macOS and windows at least as we're beholden to the local environment for those builds. For linux jobs it doesn't matter because we're running the builds in docker.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/10999/commits/6963092f351540308d3b1c3bf7bd0464f698c5c1